packages/core/src/i18n/i18nUtil.ts

```diff
@@ -8,7 +8,10 @@ export const getI18nKeyPrefixBySchema = (
   schema: i18nJsonSchema | undefined,
   uischema: UISchemaElement | undefined
 ): string | undefined => {
-  return uischema?.options?.i18n ?? schema?.i18n ?? undefined;
+  if (uischema && isControlElement(uischema) && uischema.i18n) {
```

Comment:
At the moment we only support it for `ControlElement`s but for this implementation here I would generalize this a bit. For example via an interface `Internationalizable` which is extended by `ControlElement`.

The function would then work unchanged for all UI Schema elements of the future instead of only for `Control`s.